### PR TITLE
docs: initial work on README

### PR DESCRIPTION
**Pull request by frederickfogerty**
This PR brings the library's documentation up to the current state of development.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,120 +1,422 @@
...
+https://.../image.jpg?q=35&w=100&dpr=3 3x,
+https://.../image.jpg?q=23&w=100&dpr=4 4x,
+https://.../image.jpg?q=20&w=100&dpr=5 5x
+```
+
+#### Lazy-Loading
+
+For lazy loading, there are a few options to choose from. They're listed here, along with our recommendation at the end.
+
+1. Native lazy-loading with `loading="lazy"`. As of May 2020, this is shipped in stable versions of Chrome and Firefox. [Example](#lazy-loading-native)
```
Comment (sherwinski) [inline README.md:229]: > [Example](#lazy-loading-native)
Looks like this section is missing from the README

File: README.md [hunk 2]
```diff
@@ -1,120 +1,422 @@
...
+loading="lazy" ></ix-img>
+```
+
+##### Lazy-loading (Interaction Observer)
+
+Lazy loading can be done with an Interaction Observer. The best way to do this is with a library (we recommend [Lozad.js](https://apoorv.pro/lozad.js/)), but you might also want to roll your own.
+
+You could modify the recommended code above to remove the native lazy loading support in order to create a Lozad directive.
+
+Here's a [CSSTricks article](https://css-tricks.com/lozad-js-performant-lazy-loading-images/) about using Lozad.js, and [another one](https://css-tricks.com/tips-for-rolling-your-own-lazy-loading/) about rolling your own lazy loading
```
Comment (sherwinski) [inline README.md:299]:
```suggestion
Here's a [CSSTricks article](https://css-tricks.com/lozad-js-performant-lazy-loading-images/) about using Lozad.js, and [another one](https://css-tricks.com/tips-for-rolling-your-own-lazy-loading/) about rolling your own lazy loading.
```

**Review by sherwinski — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -48,3 +48,27 @@ export class AppModule {}
...
+
+Run `ng serve` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
+
+## Code scaffolding
+
+Run `ng generate component component-name` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module`.
+
+## Build
+
+Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.
```
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply CONTRIBUTING.md:62]: You're right, I'll update this.

File: README.md [hunk 4]
```diff
@@ -1,120 +1,422 @@
...
+- some `width` CSS value (e.g. `10px`, `100%`, `calc(100vw - 10px)`)
+- `height: auto` as a CSS property
+
+`./styles.css`
+
+```css
+.test-img {
+  /* These next two lines are critical for the new browser feature. */
+  width: calc(100vw - 128px);
+  height: auto; // This tells the browser to set the height of the image to what it should be, and ignore the height attribute set on the image
```
[the remark this answers is not in the document]
Comment (frederickfogerty) [reply README.md:144]: Good catch 👍

File: README.md (hunk 1 above)
Comment (frederickfogerty) [reply README.md:229]: 👀

**Pull request merged**